Objects/tupleobject.c

```diff
@@ -119,7 +113,34 @@ PyTuple_New(Py_ssize_t size)
         if (op == NULL)
             return NULL;
     }
-    for (i=0; i < size; i++)
+    return (PyObject *) op;
+}
+
+_Py_NO_INLINE static PyObject *
+tuple_new_untracked(Py_ssize_t size)
+{
+    assert(size != 0);
+
+    return tuple_new_internal(size);
+}
+
+PyObject *
+PyTuple_New(Py_ssize_t size)
+{
+    PyTupleObject *op;
+    Py_ssize_t i;
+#if PyTuple_MAXSAVESIZE > 0
+    if (size == 0 && free_list[0]) {
+        op = free_list[0];
+        Py_INCREF(op);
+#ifdef COUNT_ALLOCS
+        _Py_tuple_zero_allocs++;
+#endif
+        return (PyObject *) op;
+    }
+#endif
+    op = (PyTupleObject *)tuple_new_internal(size);
+    for (i = 0; i < size; i++)
```

Comment:
PEP 7: please add { ... }

Or maybe use memset(0) as you did in a previous PR?

Reply:
In my tests using `memset()` here has negative effect on [benchmarks](https://pyperformance.readthedocs.io/), probably we should check it another time and add a comment if it's true, but that's for another PR.

Reply:
Aha. Some compiler uses their own "builtin" functions for most common functions like memset(): https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html

Sometimes, GCC builtins are slower than glibc functions... I don't understand well why such "non trivial" loop (involving pointer dereference) is more efficient when well defined memset(). That's the mystery of compilers :-)